### PR TITLE
Replaced the two innerHTML writes in setStackedName(...) with safe DOM-node construction

### PR DESCRIPTION
**Pull request by kelly-griffin**
### Summary:

- Replace the two innerHTML writes in setStackedName(...) with safe DOM-node construction (createElement + textContent) so no HTML is parsed. Output/markup stays identical (<span class="first">…</span><span class="last">…</span>).

Keep the existing entity logic elsewhere untouched.

Add CodeQL annotations beside the three img.src = … assignments in the image-fallback block to document that these are attribute assignments (not HTML injection).

### Files:

assets/js/statistics.js

In setStackedName(...):

Replaced

nameEl.innerHTML = `<span class="first">${esc(first)}</span><span class="last">${esc(last)}</span>`; // and
nameEl.innerHTML = `<span class="last">${esc(first || last)}</span>`;

with DOM-node construction:

nameEl.textContent = '';
// create <span.first> and <span.last> with .textContent, then append

In image fallback function:

Added one-line comments above each img.src = …:

// codeql[js/xss-through-dom]: assigning URL to  attribute; not HTML parsing. img.src = String(sources[idx++]); // and for remote1/remote2

### Risk:

- Low

Visual markup remains the same; CSS selectors targeting .first / .last continue to work.

No behavioral change to image loading; only explanatory comments added.

Edge cases verified: single-word names (only last), names with quotes/ampersands render as text (via .textContent).

### Test:

**1. Functional UI**

Load statistics.php → Skaters/Goalies/Defense blocks populate normally.

Verify top row cards show stacked name with same styling.

Click tabs (Skaters → Goalies → Teams) and confirm lists update.

**2. Images**

Confirm mugs/logos render. Disable first source to trigger fallback; image still appears via next img.src.

**3. Console**

Open DevTools → Console: no errors/warnings from statistics.js.

**4. Security**

Grep locally:
git grep -nI -E 'innerHTML|insertAdjacentHTML|outerHTML' -- assets/js/statistics.js → should show no innerHTML in setStackedName.

Push to the branch scanned by CodeQL (usually main) and confirm alerts for statistics.js close under Security → Code scanning alerts.

**5. Regression spot checks**

Names with special chars: e.g., O’Reilly, McDavid, Cale Makar render correctly.

Single-name players or missing first/last still produce one <span class="last">…</span>.

### CODE REVIEW

File: assets/js/statistics.js [hunk 1]
```diff
@@ -15,6 +15,25 @@
...
+  try {
+    const x = new URL(String(u), window.location.href); // supports relative
+    return x.protocol === 'http:' || x.protocol === 'https:';
+  } catch {
+    return false;
+  }
+};
+
+  // Resolve project asset paths relative to the current page (ignores <base>)
+  const assetURL = (p) => {
```
[the remark this answers is not in the document]
Comment (kelly-griffin) [reply assets/js/statistics.js:28]: // Pure string replacement; no DOM parsing or HTML interpretation.
// codeql[js/xss-through-dom]  lgtm[js/xss-through-dom]
const decodeEntities = (s) => String(s ?? '').replace(/&(amp|#38);/g, '&');

File: assets/js/statistics.js [hunk 2]
```diff
@@ -169,26 +196,30 @@
...
           idx++;
-          img.src = remote1;
+          // codeql[js/xss-through-dom]: league-controlled URL; still not HTML parsing.
+          img.src = String(remote1);
           img.onerror = tryNext;
         } else {
           img.onerror = null;
-          img.src = remote2;
+          // codeql[js/xss-through-dom]: final fallback URL; attribute assignment only.
+          img.src = String(remote2);
```
[the remark this answers is not in the document]
Comment (kelly-griffin) [reply assets/js/statistics.js:243]: Fixed

File: assets/js/statistics.js [hunk 3]
```diff
@@ -169,26 +196,30 @@
...
+        add(`${base}${cleanTeam}/${id}.png`);
       }
       const remote1 = mugsUrl(team, id);
       const remote2 = cmsUrl(id);
       let idx = 0;
       const tryNext = () => {
         if (idx < sources.length) {
-          img.src = sources[idx++];
+          // codeql[js/xss-through-dom]: assigning a URL to an <img> attribute; not parsing HTML.
+          img.src = String(sources[idx++]);
```
[the remark this answers is not in the document]
Comment (kelly-griffin) [reply assets/js/statistics.js:232]: Fixed

File: assets/js/statistics.js [hunk 4]
```diff
@@ -169,26 +196,30 @@
...
           img.onerror = tryNext;
         } else if (idx === sources.length) {
           idx++;
-          img.src = remote1;
+          // codeql[js/xss-through-dom]: league-controlled URL; still not HTML parsing.
+          img.src = String(remote1);
```
[the remark this answers is not in the document]
Comment (kelly-griffin) [reply assets/js/statistics.js:237]: Fixed.
